tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -125,24 +125,29 @@ private void parseArguments(final String[] args) throws IOException {
         }
     }
 
-    private void resetSourceTopicOffsets() {
+    private void resetSourceAndInternalTopicOffsets() {
         final List<String> sourceTopics = this.options.valuesOf(sourceTopicsOption);
 
         if (sourceTopics.size() == 0) {
-            System.out.println("No source topics specified, skipping resetting source topic offsets.");
-            return;
+            System.out.println("No source topics specified.");
+        } else {
+            System.out.println("Resetting offsets to zero for source topics " + sourceTopics + " and all internal topics.");
         }
 
-        System.out.println("Resetting offsets to zero for topics " + sourceTopics);
-
         final Properties config = new Properties();
         config.putAll(this.consumerConfig);
         config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, this.options.valueOf(bootstrapServerOption));
         config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, this.options.valueOf(applicationIdOption));
         config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
 
-        for (final String topic : sourceTopics) {
-            if (this.allTopics.contains(topic)) {
+        for (final String srcTopic : sourceTopics) {
+            if (!this.allTopics.contains(srcTopic)) {
+                System.out.println("Source topic " + srcTopic + " not found. Skipping.");
+            }
+        }
+
+        for (final String topic : this.allTopics) {
+            if (sourceTopics.contains(topic) || isInternalTopic(topic)) {
```

Comment:
Nit/suggestion: Use a helper method like `shouldResetOffsets(String topic)` instead of `(sourceTopics.contains(topic) || isInternalTopic(topic))`, which also clarifies the purpose of the condition.  I was initially confused why there were two for-loops working on `sourceTopics`.